SocialCareCaseViewerApi/V1/Boundary/Requests/CreateCaseStatusRequest.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using FluentValidation;
+using SocialCareCaseViewerApi.V1.Domain;
+
+namespace SocialCareCaseViewerApi.V1.Boundary.Requests
+
+#nullable enable
+{
+    public class CreateCaseStatusRequest
+    {
+        [JsonPropertyName("personId")]
+        public long PersonId { get; set; }
+
+        [JsonPropertyName("type")]
+        public string Type { get; set; } = null!;
+        [JsonIgnore]
+        public long? TypeId { get; set; }
+
+        [JsonPropertyName("fields")]
+        public List<CaseStatusRequestField>? Fields { get; set; } = new List<CaseStatusRequestField>() { }!;
+
+        [JsonPropertyName("startDate")]
+        public DateTime StartDate { get; set; }
+
+        [JsonPropertyName("endDate")]
+        public DateTime? EndDate { get; set; }
+
+        [JsonPropertyName("notes")]
+        public string Notes { get; set; } = null!;
+
+        [JsonPropertyName("createdBy")]
+        public string CreatedBy { get; set; } = null!;
+    }
+
+    public class CaseStatusRequestField
+    {
+        public String? Name { get; set; } = null!;
+        public String? Selected { get; set; } = null!;
+    }
+
+    public class CreateCaseStatusRequestValidator : AbstractValidator<CreateCaseStatusRequest>
+    {
+        public CreateCaseStatusRequestValidator()
+        {
+            RuleFor(pr => pr.PersonId)
+                .GreaterThanOrEqualTo(1).WithMessage("'personId' must be provided.");
+            RuleFor(pr => pr.Type)
+                .NotNull().WithMessage("'type' must be provided.");
+
+            RuleForEach(pr => pr.Fields)
+                .ChildRules(field =>
+                {
+                    field.RuleFor(t => t.Name).NotNull().WithMessage("Field must have a name");
+                    field.RuleFor(t => t.Selected).NotNull().WithMessage("Field selected value must not be empty");
+                });
+
+            RuleFor(x => x.StartDate)
+                .LessThan(DateTime.Now).WithMessage("'start_date' must be in the past");
+
```

Comment:
Does EndDate need to be in the future?

What about a rule ensuring EndDate must be grater than StartDate?

Reply:
Hi John,

There's no need for End Date as of now, it's going to be added/evaluated in the next iterations.
It now has been removed from the POST request